File: backend/src/Saturdaze.Application/Contracts/RestaurantDto.cs
using Saturdaze.Domain.Enums;

namespace Saturdaze.Application.Contracts;

public sealed record RestaurantDto(
    Guid Id,
    string Name,
    string Style,
    MealSlot Slot,
    bool WifeApproved,
    int DriveMinutes,
    string Notes,
    string MenuUrl,
    IReadOnlyList<RestaurantVoteDto> Votes,
    bool Locked);

public sealed record RestaurantVoteDto(string VoterName, string Vote);

File: frontend/projects/api/src/lib/models/restaurant.dto.ts
/**
 * Restaurant Dto.
 */
export interface RestaurantDto {
  /**
   * Id.
   */
  readonly id: string;
  /**
   * Name.
   */
  readonly name: string;
  /**
   * Style.
   */
  readonly style: string;
  /**
   * Slot.
   */
  readonly slot: 'Lunch' | 'Dinner';
  /**
   * Wife Approved.
   */
  readonly wifeApproved: boolean;
  /**
   * Drive Minutes.
   */
  readonly driveMinutes: number;
  /**
   * Notes.
   */
  readonly notes: string;
  /**
   * Menu Url.
   */
  readonly menuUrl?: string | null;
  /**
   * Votes.
   */
  readonly votes?: ReadonlyArray<{
    readonly voterName: string;
    readonly vote: 'up' | 'down' | 'none';
  }> | null;
  /**
   * Locked.
   */
  readonly locked?: boolean;
}
